examples/Migration/CampaignManagement/CreateCompleteCampaignGoogleAdsApiOnly.php

```diff
@@ -0,0 +1,499 @@
+            'campaign' => $campaignResourceName,
+            'status' => AdGroupStatus::ENABLED,
+            'type' => AdGroupType::SEARCH_STANDARD,
+            'cpc_bid_micros' => new Int64Value(['value' => 10000000])
+        ]);
+
+        // Creates an ad group operation.
+        $adGroupOperation = new AdGroupOperation();
+        $adGroupOperation->setCreate($adGroup);
+
+        // Issues a mutate request to add the ad groups.
+        $adGroupServiceClient = $googleAdsClient->getAdGroupServiceClient();
+        $adGroupResponse = $adGroupServiceClient->mutateAdGroups($customerId, [$adGroupOperation]);
+
+        $adGroupResourceName = $adGroupResponse->getResults()[0]->getResourceName();
+        $newAdGroup = self::getAdGroup($googleAdsClient, $customerId, $adGroupResourceName);
+
+        printf("Added ad group named '%s'%s", $newAdGroup->getName()->getValue(), PHP_EOL);
+
+        return $newAdGroup;
+    }
+
+    /**
+     * Gets a campaign.
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param string $customerId the client customer ID without hyphens
+     * @param string $resourceName the resource name of the ad group to retrieve
+     * @return AdGroup the ad group
+     */
+    private static function getAdGroup(
+        GoogleAdsClient $googleAdsClient,
+        string $customerId,
+        string $adGroupResourceName
+    ) {
+        $googleAdsServiceClient = $googleAdsClient->getGoogleAdsServiceClient();
+        $query = "SELECT ad_group.id, ad_group.name, " .
+                 "ad_group.resource_name from ad_group where " .
+                 "ad_group.resource_name='$adGroupResourceName'";
+
+        $response =
+            $googleAdsServiceClient->search($customerId, $query, ['pageSize' => self::PAGE_SIZE]);
+
+        return $response->getIterator()->current()->getAdGroup();
+    }
+
+    /**
+     * Creates text ads.
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param string $customerId the client customer ID without hyphens
+     * @param AdGroup $adGroup the ad group
+     * @return AdGroupAd[] an array of text ads
+     */
+    private static function createTextAds(
+        GoogleAdsClient $googleAdsClient,
+        string $customerId,
+        AdGroup $adGroup
+    ) {
+        $adGroupResourceName = $adGroup->getResourceName();
+
+        $operations = [];
+        for ($i = 0; $i < self::NUMBER_OF_ADS; $i++) {
+            // Creates the expanded text ad info.
+            $expandedTextAdInfo = new ExpandedTextAdInfo([
+                'headline_part1' => new StringValue(['value' => 'Cruise to Mars #' . uniqid()]),
+                'headline_part2' => new StringValue(['value' => 'Best Space Cruise Line']),
+                'description' => new StringValue(['value' => 'Buy your tickets now!'])
+            ]);
+
+            // Sets the expanded text ad info on an Ad.
+            $ad = new Ad([
```

Comment:
Inline Ad and ExpandedTextAdInfo would be too nested.
But at least I would inline either ExpandedTextAdInfo into Ad or Ad into AdGroupAd.
(I prefer the latter)

Reply:
Done.